lib/ansible/module_utils/vmware.py

```diff
@@ -912,80 +912,83 @@ def get_vm(self):
         vm_obj = None
         user_desired_path = None
         use_instance_uuid = self.params.get('use_instance_uuid') or False
-        if self.params['uuid'] and not use_instance_uuid:
-            vm_obj = find_vm_by_id(self.content, vm_id=self.params['uuid'], vm_id_type="uuid")
-        elif self.params['uuid'] and use_instance_uuid:
-            vm_obj = find_vm_by_id(self.content,
-                                   vm_id=self.params['uuid'],
-                                   vm_id_type="instance_uuid")
-        elif self.params['name']:
-            objects = self.get_managed_objects_properties(vim_type=vim.VirtualMachine, properties=['name'])
-            vms = []
-
-            for temp_vm_object in objects:
-                if len(temp_vm_object.propSet) != 1:
-                    continue
-                for temp_vm_object_property in temp_vm_object.propSet:
-                    if temp_vm_object_property.val == self.params['name']:
-                        vms.append(temp_vm_object.obj)
-                        break
-
-            # get_managed_objects_properties may return multiple virtual machine,
-            # following code tries to find user desired one depending upon the folder specified.
-            if len(vms) > 1:
-                # We have found multiple virtual machines, decide depending upon folder value
-                if self.params['folder'] is None:
-                    self.module.fail_json(msg="Multiple virtual machines with same name [%s] found, "
+        if 'uuid' in self.params:
+            if self.params['uuid'] and not use_instance_uuid:
+                vm_obj = find_vm_by_id(self.content, vm_id=self.params['uuid'], vm_id_type="uuid")
+            elif self.params['uuid'] and use_instance_uuid:
+                vm_obj = find_vm_by_id(self.content,
+                                       vm_id=self.params['uuid'],
+                                       vm_id_type="instance_uuid")
+        if 'name' in self.params:
+            if self.params['name']:
```

Comment:
This check is redundant since we have `if 'name'...` on previous line.

Reply:
No! Actually I thought about it.  Both are different checks
First check is to check if key 'name' in Params dict.
Second check is if Value is assigned to params['name']. Here params['name'] can be None

Reply:
```python
if 'name' in self.params and self.params['name']:
```